--- a/lib/src/ffi.dart	2021-10-04 21:58:27.000000000 +0300
+++ b/lib/src/ffi.dart	2026-09-13 04:50:27.481651700 +0300
@@ -186,7 +186,7 @@
 
 final Map<Pointer<JSRuntime>, _RuntimeOpaque> runtimeOpaques = Map();
 
-Pointer<JSValue>? channelDispacher(
+Pointer<JSValue> channelDispacher(
   Pointer<JSContext> ctx,
   int type,
   Pointer<JSValue> argv,
@@ -194,7 +194,7 @@
   final rt = type == JSChannelType.FREE_OBJECT
       ? ctx.cast<JSRuntime>()
       : jsGetRuntime(ctx);
-  return runtimeOpaques[rt]?._channel(ctx, type, argv);
+  return runtimeOpaques[rt]?._channel(ctx, type, argv) ?? nullptr.cast<JSValue>();
 }
 
 Pointer<JSRuntime> jsNewRuntime(
@@ -202,7 +202,7 @@
   int timeout,
   ReceivePort port,
 ) {
-  final rt = _jsNewRuntime(Pointer.fromFunction(channelDispacher), timeout);
+  final rt = _jsNewRuntime(Pointer.fromFunction<_JSChannelNative>(channelDispacher), timeout);
   runtimeOpaques[rt] = _RuntimeOpaque(callback, port);
   return rt;
 }
